### PR TITLE
Move jsdom to devDependencies

### PR DESCRIPTION
**Pull request by Woyken**
## Changes

Package should not include dependencies that are not used during runtime

## Testing

Unit tests pass

## Docs

"N/A"

### CODE REVIEW

File: packages/itwinui-react/package.json [hunk 1]
```diff
@@ -80,10 +80,8 @@
     "@floating-ui/react": "^0.26.3",
     "@itwin/itwinui-illustrations-react": "^2.1.0",
     "classnames": "^2.3.2",
-    "jsdom": "^24.0.0",
     "react-table": "^7.8.0",
-    "react-transition-group": "^4.4.5",
-    "tslib": "^2.6.0"
```
Comment (mayank99) [inline packages/itwinui-react/package.json:86]: Could you undo this? `tslib` is intentionally made part of deps. It helps make the compiled `tsc` output smaller

Comment (Woyken) [reply packages/itwinui-react/package.json:86]: Oh, okay, got it, I searched for import references, there were none, I thought it was an accident.
Reverted

**Review by mayank99 — APPROVED** [review]
LGTM, thanks for the PR

**Comment by mayank99** [comment]
Released in [3.4.2](https://github.com/iTwin/iTwinUI/releases/tag/%40itwin%2Fitwinui-react%403.4.2) 🚀

**Pull request merged**